src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -159,15 +159,21 @@ private static AnalyzerLoadFailureEventArgs CreateAnalyzerFailedArgs(Exception e
         /// </summary>
         /// <exception cref="BadImageFormatException">The PE image format is invalid.</exception>
         /// <exception cref="IOException">IO error reading the metadata.</exception>
+        [PerformanceSensitive( "https://github.com/dotnet/roslyn/issues/30449" )]
         private static ImmutableDictionary<string, ImmutableHashSet<string>> GetAnalyzerTypeNameMap(string fullPath, AttributePredicate attributePredicate)
         {
             using (var assembly = AssemblyMetadata.CreateFromFile(fullPath))
             {
+                // This is longer than strictly necessary to avoid thrashing the GC with string allocations
+                // in the call to GetFullyQualifiedTypeNames. Specifically, this checks for the presence of
+                // supported languages prior to creating the type names.
                 var typeNameMap = from module in assembly.GetModules()
                                   from typeDefHandle in module.MetadataReader.TypeDefinitions
-                                  let typeDef = module.MetadataReader.GetTypeDefinition(typeDefHandle)
-                                  let typeName = GetFullyQualifiedTypeName(typeDef, module.Module)
-                                  from supportedLanguage in GetSupportedLanguages(typeDef, module.Module, attributePredicate)
+                                  let typeDef = module.MetadataReader.GetTypeDefinition( typeDefHandle )
```

Comment:
:memo: this style change came through because #22143 hasn't been merged yet, so some local settings are able to override the general team style. The deviation will be automatically corrected by the upcoming formatting analyzer if it's not updated in this pull request before merging.